### PR TITLE
Week 3 -> week 3 exercise with the details 

### PR DESCRIPTION
**Pull request by vd89**
@anilrao316 

The code is done for the review

### CODE REVIEW

File: Week-3/exercise_3.1/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+memorizeAdd(100,100);
+timeEnd("process_1");
+time("process_2");
+memorizeAdd(100,200);
+timeEnd("process_2");
+time("process_3");
+memorizeAdd(100,100);
+timeEnd("process_3");
+time("process_4");
+memorizeAdd(100,200);
```
Comment (anilrao316) [inline Week-3/exercise_3.1/index.js:34]: if we change the order of parameters as in memorizeAdd(200,100) will it return from cache for the second time? if not we have handle that case as well

**Comment by anilrao316** [comment]
Add recording of working solution as part of this PR

**Comment by anilrao316** [comment]
Rest looks good!

File: Week-3/exercise_3.1/index.js (hunk 1 above)
Comment (vd89) [reply Week-3/exercise_3.1/index.js:34]: With the desired change it looks good

**Comment by anilrao316** [comment]
Looks good we are good to merge

**Comment by anilrao316** [comment]
> Add recording of working solution as part of this PR

PLease add this

**Pull request merged**